go-controller/pkg/util/kube.go

```diff
@@ -65,12 +66,20 @@ func IsClusterIPSet(service *kapi.Service) bool {
 	return service.Spec.ClusterIP != kapi.ClusterIPNone && service.Spec.ClusterIP != ""
 }
 
+// ValidatePort checks if the port is non-zero and port protocol is valid
+func ValidatePort(proto kapi.Protocol, port int32) error {
```

Comment:
should you also check upper limit of 65535?

Reply:
Yes, will do